src/kps/business/RouteCalculator.java

```diff
@@ -10,12 +10,13 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.stream.Collectors;
 
 public class RouteCalculator {
     private List<Route> routes;
 
     public RouteCalculator(Simulation simulation) {
-        this.routes = simulation.getRoutes();
+        this.routes = simulation.getUniqueRoutes().stream().filter(r -> !r.isDiscontinued()).collect(Collectors.toList());
```

Comment:
Its a confusing loop. Ill try to write some psudo code:
for all routes as route1:
  for all routes as route2:
    if route1 is route 2 then continue
    if route1 from and to are not the same as route2's then continue (this line was missing and broke it)
    if route 1 was created before route 2 then ignore this route
  end
end